src/backtracking/all_combination_of_size_k.rs

```diff
@@ -1,62 +1,121 @@
-/*
-    In this problem, we want to determine all possible combinations of k
-    numbers out of 1 ... n. We use backtracking to solve this problem.
-    Time complexity: O(C(n,k)) which is O(n choose k) = O((n!/(k! * (n - k)!)))
-
-    generate_all_combinations(n=4, k=2) => [[1, 2], [1, 3], [1, 4], [2, 3], [2, 4], [3, 4]]
-*/
-pub fn generate_all_combinations(n: i32, k: i32) -> Vec<Vec<i32>> {
-    let mut result = vec![];
-    create_all_state(1, n, k, &mut vec![], &mut result);
-
-    result
+//! This module provides a function to generate all possible combinations
+//! of `k` numbers out of `1...n` using a backtracking algorithm.
```

Comment:
I think it would be more natural to have:
```suggestion
//! of `k` numbers out of `0...n-1` using a backtracking algorithm.
```

What do you think?

Reply:
Yes, It would be more natural when using `0...n-1` instead of `1...n`. Especially since it's consistent with how ranges are commonly used in programming languages.